src/App.js

```diff
@@ -1,77 +1,183 @@
-  const [url, setUrl] = useState('')
+  const [balance, setBalance] = useState(0);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isComputerLoading, setIsComputerLoading] = useState(true);
 
-  const [revs, setRevs] = useState([])
-  const [artworks, setArtworks] = useState([])
-  const [refresh, setRefresh] = useState(0)
+  const [title, setTitle] = useState("");
+  const [artist, setArtist] = useState("");
+  const [url, setUrl] = useState("");
 
-  useEffect(() => {
-    const fetchRevs = async () => {
-      setBalance(await computer.db.wallet.getBalance())
-      setRevs(await computer.getRevs(computer.db.wallet.getPublicKey()))
-      setTimeout(() => setRefresh(refresh + 1), 3500)
-    }
-    fetchRevs()
-  }, [computer.db.wallet, refresh])
+  // const [revs, setRevs] = useState([]);
+  const [artworks, setArtworks] = useState([]);
+  // const [refresh, setRefresh] = useState(0);
 
-  useEffect(() => {
-    const fetchArtworks = async () => {
-      setArtworks(await Promise.all(revs.map(async rev => computer.sync(rev))))
-    }
-    fetchArtworks()
-  }, [revs, computer])
+  function useInterval(callback, delay) {
+    const savedCallback = useRef();
 
-  useEffect(() => console.log('revs', revs), [revs])
-  useEffect(() => console.log('artworks', artworks), [artworks])
+    // Remember the latest callback.
+    useEffect(() => {
+      savedCallback.current = callback;
+    }, [callback]);
 
-  const handleSubmit = async (evt) => {
-    evt.preventDefault()
-    const artwork = await computer.new(Artwork, [title, artist, url])
-    console.log('created artwork', artwork)
+    // Set up the interval.
+    useEffect(() => {
+      function tick() {
+        savedCallback.current();
+      }
+      if (delay !== null) {
+        let id = setInterval(tick, delay);
+        return () => clearInterval(id);
+      }
+    }, [delay]);
   }
 
-  return (
-    <div className="App">
-      <h2>Wallet</h2>
-      <b>Address</b>&nbsp;{computer.db.wallet.getAddress().toString()}<br />
-      <b>Public Key</b>&nbsp;{computer.db.wallet.getPublicKey().toString()}<br />
-      <b>Balance</b>&nbsp;{balance/1e8} {computer.db.wallet.restClient.chain}<br />
-      <button type="submit" onClick={() => setComputer(new Computer())}>
-        Generate New Wallet
-      </button>
+  useEffect(() => {
+    console.log("new address generated");
+    setIsComputerLoading(true);
+  }, [computer]);
 
-      <h2>Create new Artwork</h2>
-      <form onSubmit={handleSubmit}>
-        Title<br />
-        <input type="string" value={title} onChange={e => setTitle(e.target.value)} />
+  useInterval(async () => {
+    try {
+      if (computer) {
+        const newBalance = await computer.db.wallet.getBalance();
+        const newRevs = await computer.getRevs(
+          computer.db.wallet.getPublicKey()
+        );
+        const newArts = await Promise.all(
+          newRevs.map(async (rev) => computer.sync(rev))
+        );
```

Comment:
Sync is a potentially slow operation as it needs to do a recursive analysis of the blockchain. We want to minimize the number of calls to "sync".

One way to do it is to initialize two arrays "revs" and "artworks" to empty arrays. Then in a loop:
* use getRevs to query for the latest revs (call that array something line newRevs)
* then compute "deleted revs" and "new revs" by comparing the newly computed revs to latest revs
* remove the artworks in "deleted revs" from artworks
* use sync to compute the new artworks and add them to the "artworks" array